JavaScript/Topics/Flavours/README.md

```diff
@@ -0,0 +1,115 @@
+# JSX:-
+
+JSX (JavaScript Syntax Extension and occasionally referred as JavaScript XML) is an extension to the JavaScript language syntax[1] which provides a way to structure component rendering using syntax familiar to many developers. It is similar in appearance to HTML.
+
+React components are typically written using JSX, although they do not have to be as components may also be written in pure JavaScript. 
+
+## Why JSX:-
+
+React embraces the fact that rendering logic is inherently coupled with other UI logic: how events are handled, how the state changes over time, and how the data is prepared for display.
+
+Instead of artificially separating technologies by putting markup and logic in separate files, React separates concerns with loosely coupled units called “components” that contain both. We will come back to components in a further section, but if you’re not yet comfortable putting markup in JS, this talk might convince you otherwise.
+
+React doesn’t require using JSX, but most people find it helpful as a visual aid when working with UI inside the JavaScript code. It also allows React to show more useful error and warning messages.
+
+**Coding JSX:-**
+
+JSX allows us to write HTML elements in JavaScript and place them in the DOM without any _createElement()_  and/or _appendChild()_ methods.
+
+JSX converts HTML tags into react elements.
+
+You are not required to use JSX, but JSX makes it easier to write React applications.
+
+Here are two examples. The first uses JSX and the second does not:
+
+Example 1
+**JSX:**
+
```

Comment:
Write like this:
```jsx
const myelement = <h1>I Love JSX!</h1>;
< ReactDOM.render(myelement, document.getElementById('root'));
```

```
```jsx
const myelement = <h1>I Love JSX!</h1>;
< ReactDOM.render(myelement, document.getElementById('root'));
`` `
```